src/pkg/cli/composeStart.go

```diff
@@ -18,6 +22,21 @@ func ComposeStart(ctx context.Context, c client.Client, project *compose.Project
 	if err := validateProject(project); err != nil {
 		return nil, &ComposeError{err}
 	}
+
+	// Create a regexp to detect private service names in environment variable values
+	var serviceNames []string
+	for _, svccfg := range project.Services {
+		if isPrivate(&svccfg) && slices.ContainsFunc(svccfg.Ports, func(p compose.ServicePortConfig) bool {
+			return p.Mode == "host" // only private services with host ports get DNS names
+		}) {
+			serviceNames = append(serviceNames, regexp.QuoteMeta(svccfg.Name))
+		}
+	}
+	var serviceNameRegex *regexp.Regexp
+	if len(serviceNames) > 0 {
+		serviceNameRegex = regexp.MustCompile(`\b(?:` + strings.Join(serviceNames, "|") + `)\b"`)
```

Comment:
Compose spec [allows '.' to be part of the service name](https://github.com/compose-spec/compose-spec/blob/master/schema/compose-spec.json#L33), does it mean we will construct regex like `a.b|c.d` which matches not only a.b but also aab, axb etc?
What is the final double quote after \b is for?

Reply:
* I'm using `regexp.QuoteMeta` to escape regex stuff within the names
* last quote is an error; will add a test!